### PR TITLE
Improve documentation

### LINKED ISSUE #25: Fix documentation
Two step process:
1. Fix documentation pipeline & publishing - @c0derMo 
2. Update in code documentation: All contributors

### PR DESCRIPTION
**Pull request by c0derMo**
Closes #25.

This PR aims to improve code quality by adding more linting rules, aswell as enforcing jsdoc comments.

This PR also adds automatic documentation builds & publishes them to https://namelessgroup.github.io/namelessbot
The published version only updates when triggered manually or at pushes to master.

Reviews from all three contributors (@c0derMo, @Kr0nox, @MartinKrausewitz) should be awaited before merging this request.

### CODE REVIEW

**Review by MartinKrausewitz — APPROVED** [review]
Seems pretty good. We need to force @myphyk to implement the wednesday recurring message ;D

File: lib/truthtable.ts [hunk 1]
```diff
@@ -483,6 +758,12 @@ interface TruthTable {
     results: { assignment: boolean[]; result: boolean; }[];
 }
 
+/**
+ * Generates a truthtable with the supplied AST & variables
```
Comment (Kr0nox) [inline lib/truthtable.ts:762]: Should we use abbreviations that may not be known to all developers like ast? or should we use the written out from?

File: slashCommands/aoc.ts [hunk 2]
```diff
@@ -3,6 +3,9 @@ import {CommandInteraction} from "discord.js";
 import {embedLeaderboard} from "../lib/aocleaderboardparser";
 import {get} from "../lib/configmanager";
 
+/**
+ * Slash command definition for /aoc, a command that sends the current leaderboard
```
Comment (Kr0nox) [inline slashCommands/aoc.ts:7]: All Slash-Command-Docs have the same structure except timetable (Wirrten like it is here). Should we make that uniform.

File: tasks/aoc.ts [hunk 3]
```diff
@@ -1,9 +1,16 @@
 import {Client, TextChannel} from "discord.js";
 import {TaskExecutor} from "../types";
-import {embedLeaderboard} from "../lib/aocleaderboardparser"
+import {embedLeaderboard} from "../lib/aocleaderboardparser";
 import {get} from "../lib/configmanager";
 import {DateTime} from "luxon";
 
+/**
+ * TaskExecutor that sends the current standings for the AoC competition.
```
Comment (Kr0nox) [inline tasks/aoc.ts:8]: Should we also make the documentation here uniform with the others in their wording at the start?

**Review by Kr0nox — CHANGES_REQUESTED** [review]

**Comment by Kr0nox** [comment]
And @c0derMo we dscussed having the README on the first page of the docs. Is that possible?

**Comment by c0derMo** [comment]
> And @c0derMo we dscussed having the README on the first page of the docs. Is that possible?

Should already be the case - the currently deployed version of the docs is *not* up to date. To look at the result, build them locally using `yarn doc`

**Comment by c0derMo** [comment]
Also, we could concider getting rid of "messageCommands", since we don't use them since the introduction of slash commands

**Comment by c0derMo** [comment]
Another question would be, if we want to enforce the usage of "full sentences" in description `@param` and '@returns` tags - "full sentences" being classified as ones starting with uppercase letters, ending in period, question mark, exclamation mark, etc.
@Kr0nox @MartinKrausewitz

**Comment by Kr0nox** [comment]
Removing is is definitely the right way to go. Would it be better to remove it in this PR or when restructuring the entire code into modules?

**Review by c0derMo — COMMENTED** [review]
lgtm - apart from the open question about "chatCommands" (commands using messages starting with `!`)

**Comment by c0derMo** [comment]
> Removing is is definitely the right way to go. Would it be better to remove it in this PR or when restructuring the entire code into modules?

@Kr0nox - I think we can also do this in the upcoming PR about restructuring, you're right
In which case, only your review is left until we can merge this ;)

**Review by Kr0nox — APPROVED** [review]
Everything lgtm

**Pull request merged**